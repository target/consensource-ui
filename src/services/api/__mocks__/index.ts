export * from './factory';
export * from './certificate';
