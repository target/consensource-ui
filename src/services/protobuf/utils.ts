import {
  CertificateRegistryPayload,
  ICertificateRegistryPayload,
} from './compiled';

export interface PayloadInfo {
  payloadBytes: string | Buffer | NodeJS.TypedArray | DataView;
  inputs: string[];
  outputs: string[];
}

export const ACTIONS = CertificateRegistryPayload.Action;

/**
 * Encodes a CertificateRegistryPayload message
 * @param message CertificateRegistryPayload message or plain object to encode
 */
export function encodePayload(
  payload: ICertificateRegistryPayload,
): Uint8Array {
  return CertificateRegistryPayload.encode(payload).finish();
}
