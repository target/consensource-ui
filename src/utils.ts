// Util collection

/**
 * Selects keys from an object
 */
export const pluck = (obj: any, ...keys: Array<any>): any =>
    keys.reduce((o, k) => {
        if (typeof obj[k] !== 'undefined') {
            o[k] = obj[k];
        }
        return o;
    }, {});

/**
 * Merges objects, starting from left to right
 */
