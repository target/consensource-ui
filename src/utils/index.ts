export function getLocaleFromUnix(timestampSeconds: number) {
  const timestampMS = timestampSeconds * 1000;
  return new Date(timestampMS).toLocaleDateString();
}
