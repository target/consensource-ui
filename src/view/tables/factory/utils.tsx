import { MUIDataTableOptions, MUIDataTableChip } from 'mui-datatables';

export const DEFAULT_ROWS_PER_PAGE = 100;

export const textLabels: MUIDataTableOptions['textLabels'] = {
  body: {
    noMatch: 'No factories found',
  },
};

/**
 * Default props for the filter chips displayed in the
 * `<TableFilterList />`.
 */
export const filterChipProps: MUIDataTableChip = {
  color: 'secondary',
  variant: 'default',
};

/**
 * Utility function to convert a string to an array, or return
 * the original param if an array was passed.
 *
 * This operation is frequent when decoding URL query params.
 */
export const convertStrToArray = (val: string[] | string) => {
  return typeof val === 'string' ? [val] : val;
};
