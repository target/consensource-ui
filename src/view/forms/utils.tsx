import React from 'react';
import { Grid, Typography, TextFieldProps } from '@material-ui/core';
import { BatchStatusRes } from 'services/api';

export interface TransactionFormProps {
  setBatchStatusLink: (statusLink: BatchStatusRes['link']) => void;
}

export type onChangeEvent = Parameters<
  NonNullable<TextFieldProps['onChange']>
>[0];

/**
 * Helper function that checks if any values in `state` have not been set.
 * Returns true if there is any key in `state` that is falsey.
 */
export function hasEmptyFields<T>(state: T) {
  return Object.keys(state).some((key) => !state[key as keyof T]);
}

export interface FormErrMsgProps {
  msg?: string;
}

export function FormErrMsg({ msg }: FormErrMsgProps) {
  if (!msg) {
    return null;
  }

  return (
    <Grid item>
      <Typography variant="caption" color="error">
        {msg}
      </Typography>
    </Grid>
  );
}
