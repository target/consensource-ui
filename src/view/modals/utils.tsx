import { DialogProps as MUIDialogProps } from '@material-ui/core';
import { FactoryResData } from 'services/api';

export interface DialogProps extends MUIDialogProps {
  handleClose: () => void;
}

export interface OrgDialogProps extends DialogProps {
  existing_org: FactoryResData;
}
