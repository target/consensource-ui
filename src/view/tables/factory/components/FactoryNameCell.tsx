import React from 'react';
import { FactoryResData } from 'services/api';
import { Tooltip } from '@material-ui/core';
import { Link } from 'view/components';

export interface FactoryNameCellProps {
  name: FactoryResData['name'];
  id: FactoryResData['id'];
}

export const FactoryNameCell = ({ name, id }: FactoryNameCellProps) => {
  return (
    <Link to={`factories/${id}`} color="secondary">
      <Tooltip placement="top" title="View factory profile">
        <i>{name}</i>
      </Tooltip>
    </Link>
  );
};
