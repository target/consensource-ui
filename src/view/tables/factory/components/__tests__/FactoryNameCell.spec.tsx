import React from 'react';
import { render, screen } from 'utils/testing';
import { mockFactoryRes } from 'services/api/__mocks__';
import userEvent from '@testing-library/user-event';
import { FactoryNameCell } from '../FactoryNameCell';

describe('<FactoryNameCell />', () => {
  describe('with factories', () => {
    const { data: FactoryData } = mockFactoryRes;

    it('renders', () => {
      const { container } = render(
        <FactoryNameCell name={FactoryData[0].name} id={FactoryData[0].id} />,
      );
      expect(container).toMatchSnapshot();
    });

    it('sets the link destination', () => {
      const { id } = FactoryData[0];

      render(
        <FactoryNameCell name={FactoryData[0].name} id={FactoryData[0].id} />,
      );

      expect(screen.getByRole('link')).toHaveAttribute(
        'href',
        `factories/${id}`,
      );
    });

    it('displays informational text in the tooltip', async () => {
      render(
        <FactoryNameCell name={FactoryData[0].name} id={FactoryData[0].id} />,
      );

      userEvent.hover(screen.getByTitle('View factory profile'));

      expect(
        await screen.findByText('View factory profile'),
      ).toBeInTheDocument();
    });
  });
});
