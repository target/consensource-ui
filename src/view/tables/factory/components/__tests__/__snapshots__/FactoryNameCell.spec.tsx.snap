// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FactoryNameCell /> with factories renders 1`] = `
<div>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
    href="/factories/3e23e8160039594a33894f6564e1b1348bbd7a0088d42c4acb73eeaed59c009d"
  >
    <i
      class=""
      title="View factory profile"
    >
      Test Org 1
    </i>
  </a>
</div>
`;
